frontend/src/scenes/persons/GroupActorHeader.tsx

```diff
@@ -0,0 +1,18 @@
+import { GroupActorType } from '~/types'
+import React from 'react'
+import { IconPersonFilled } from 'lib/components/icons'
+import './PersonHeader.scss'
+
+export interface GroupActorHeaderProps {
+    actor?: Partial<GroupActorType> | null
+    withIcon?: boolean
+}
+
+export function GroupActorHeader(props: GroupActorHeaderProps): JSX.Element {
+    return (
+        <div className="person-header identified">
+            {props.withIcon && <IconPersonFilled style={{ marginRight: 8 }} />}
+            <span className="ph-no-capture">{props.actor?.properties?.name || props.actor?.id}</span>
```

Comment:
Not sure why you went this route, this is the worst of all worlds. We're re-introducing https://github.com/PostHog/posthog/issues/7432 for another class of victims.

For persons, the existing logic is outlined here (https://github.com/PostHog/posthog/issues/7432), we should use `asDisplay`

For groups, we use group_key (actor.id) for displaying it always.